src/concrete/ml/sklearn/base.py

```diff
@@ -1328,6 +1339,266 @@ def __init__(self, n_bits: Union[int, Dict[str, int]]):
 
         BaseEstimator.__init__(self)
 
+    # pylint: disable=too-many-locals,too-many-statements,too-many-branches
+    @classmethod
+    def from_sklearn_model(
+        cls,
+        sklearn_model: sklearn.base.BaseEstimator,
+        X: Optional[numpy.ndarray] = None,
+        n_bits: int = 8,
+    ):
+        """Build a FHE-compliant model using a fitted scikit-learn model.
+
+        Args:
+            sklearn_model (sklearn.base.BaseEstimator): The fitted scikit-learn model to convert.
+            X (Optional[Data]): A representative set of input values used for computing quantization
+                parameters, as a Numpy array, Torch tensor, Pandas DataFrame or List. This is
+                usually the training data-set or a sub-set of it.
+            n_bits (int): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and weights. If a dict is
+                passed, then it should contain "op_inputs" and "op_weights" as keys with
+                corresponding number of quantization bits so that:
+                - op_inputs : number of bits to quantize the input values
+                - op_weights: number of bits to quantize the learned parameters
+                Default to 8.
+
+        Returns:
+            The FHE-compliant fitted model.
+        """
+        # Check that sklearn_model is a proper fitted scikit-learn model
+        check_is_fitted(sklearn_model)
+
+        # Extract scikit-learn's initialization parameters
+        init_params = sklearn_model.get_params()
+
+        # Instantiate the Concrete ML model and update initialization parameters
+        # This update is necessary as we currently store scikit-learn attributes in Concrete ML
+        # classes during initialization (for example: link or power attributes in GLMs)
+        # Without it, these attributes will have default values instead of the ones used by the
+        # scikit-learn models
+        # This should be fixed once Concrete ML models initialize their underlying scikit-learn
+        # models during initialization
+        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3373
+        # Needed for XGB
+
+        cls_signature = inspect.signature(cls)
+        init_params_keys = list(init_params.keys())
+        for key in init_params_keys:  # pragma: no cover
+            if key not in cls_signature.parameters:
```

Comment:
why is it mandatory to do this ? couldn't we just do `cls(n_bits=n_bits, **init_params)` like for linear models ? especially since we now check that all sklearn attributes are supported in CML (thanks to @jfrery )

Reply:
Actually this wasn't needed